cpp/src/strings/convert/convert_datetime.cu

```diff
@@ -527,6 +535,13 @@ struct datetime_formatter
                 break;
             }
         }
+
+        // compute day of the year
+        // for month >= 10, leap-year has been already been included
+        timeparts[TP_DAY_OF_YEAR] = (month >= 10) ? days - monthDayOffset[10] +1 // 306
```

Comment:
How can I find out more about what's going on here? How come October has something to do with leap years?

Reply:
The number of days is calculated here is simplified by making March the first month so that February is the last month and any extra leap days are just added to the end of the year. This shift is accounted for here and the code below here when computing the actual month, day, hours, etc.
I can add comment here to make this clearer.